pennylane_lightning/core/_serialize.py

```diff
@@ -183,19 +184,21 @@ def _named_obs(self, observable, wires_map: dict = None):
 
     def _hermitian_ob(self, observable, wires_map: dict = None):
         """Serializes a Hermitian observable"""
-        assert not isinstance(observable, Tensor)
+        assert not isinstance(observable, (Tensor, Prod))
 
         wires = [wires_map[w] for w in observable.wires] if wires_map else observable.wires.tolist()
         return self.hermitian_obs(matrix(observable).ravel().astype(self.ctype), wires)
 
     def _tensor_ob(self, observable, wires_map: dict = None):
         """Serialize a tensor observable"""
-        assert isinstance(observable, Tensor)
-        return self.tensor_obs([self._ob(obs, wires_map) for obs in observable.obs])
+        assert isinstance(observable, (Tensor, Prod))
```

Comment:
Should we raise an error here or simply `assert`?

Reply:
In my opinion, the less changes we have to make to get new operator arithmetic to work, the better. The assertions were there before, and I don't really see a reason to change them.

Reply:
As per above discussion, the assertions will be removed.